packages/grid/_modules_/grid/components/containers/GridRootStyles.ts

```diff
@@ -88,7 +88,7 @@ export const useStyles = makeStyles(
           padding: theme.spacing(0, 2),
         },
         '& .MuiDataGrid-colCell:focus, & .MuiDataGrid-cell:focus': {
-          outline: 'dotted',
+          outline: `solid ${theme.palette.primary.main} 1px`,
```

Comment:
You should use the same `:not(.MuiDataGrid-cellCheckbox)` here to not show the outline in the checkbox column. If I click in the corner it won't match the `:focus-within`.

![focus](https://user-images.githubusercontent.com/42154031/117306949-84996d00-ae56-11eb-9fe3-7c844bb577c9.gif)